app/scripts/components/exploration/container.tsx

```diff
@@ -32,6 +33,8 @@ const tourProviderStyles = {
 };
 
 export default function ExplorationAndAnalysisContainer() {
+  const setExternalDatasets = useSetAtom(externalDatasetsAtom);
+  setExternalDatasets(allExploreDatasets);
```

Comment:
The current approach using `setExternalDatasets` works, but it might need to be called in multiple places where we depend on the static datasets. 

I was wondering if a more scalable solution would be to use [our existing](https://github.com/NASA-IMPACT/veda-ui/blob/14fc8ea4dabb10a0c1912230c6e71359798ca614/app/scripts/context/env-config.tsx#L9) `EnvConfigProvider` and evolve it into a more general `ConfigProvider`. Like this it would centralize both our env and data configuration in one place.

The provider could be used like this (note the extra datasets prop passed):

```
<EnvConfigProvider
      config={{
                envMapboxToken: process.env.NEXT_PUBLIC_MAPBOX_TOKEN ?? '',
                envApiStacEndpoint: process.env.NEXT_PUBLIC_API_STAC_ENDPOINT ?? '',
                envApiRasterEndpoint: process.env.NEXT_PUBLIC_API_RASTER_ENDPOINT ?? '',
                datasets: static-datasets
      }}
>
              {children}
</EnvConfigProvider>
```

And internally, the provider would handle dataset hydration:

```
const setExternalDatasets = useSetAtom(externalDatasetsAtom);
setExternalDatasets(allExploreDatasets);
```

Like that the setter won't have to be manually called several times if it's needed.

Reply:
How you described makes sense. My concern for now is how nicely it would work out irl considering Jotai places itself as a replacement for context (I am reading https://jotai.org/docs/basics/comparison) - or it just means that we are really not taking full advantage of Jotai. I will put more thoughts into this from the perspective of global state management.